proxy/http2/Http2ConnectionState.cc

```diff
@@ -1626,10 +1621,10 @@ Http2ConnectionState::send_headers_frame(Http2Stream *stream)
   } else {
     payload_length = BUFFER_SIZE_FOR_INDEX(buffer_size_index[HTTP2_FRAME_TYPE_HEADERS]);
   }
-  Http2Frame headers(HTTP2_FRAME_TYPE_HEADERS, stream->get_id(), flags);
-  headers.alloc(buffer_size_index[HTTP2_FRAME_TYPE_HEADERS]);
-  http2_write_headers(buf, payload_length, headers.write());
-  headers.finalize(payload_length);
+
+  Http2HeadersFrame headers(stream->get_id(), flags, buf, payload_length);
+  this->ua_session->xmit(&headers);
```

Comment:
I think we should call xmit after changing the stream state, because sending this frame may cause a protocol violation if the stream is not in an appropriate state.

Reply:
I'll back the order.
It's a bit wired checking stream state by `Http2Stream::change_state()`, though.